### PR TITLE
Prod 1748 hover tooltip

### PR DESCRIPTION
**Pull request by vas3a**

### CODE REVIEW

**Review by brooketopcoder — APPROVED** [review]

File: src-ts/lib/functions/component-visible-functions/component-visible.functions.tsx [hunk 1]
```diff
@@ -1,28 +1,53 @@
...
+import { MouseEvent as RMouseEvent, MouseEventHandler, MutableRefObject, useCallback, useEffect, useRef } from 'react'
 
-export interface ComponentVisible {
-    isComponentVisible: boolean
-    ref: MutableRefObject<any>
-    setIsComponentVisible: Dispatch<SetStateAction<boolean>>
-}
+/**
+ * Registers an outside click event handler, and calls the callback on click outside
+ * @param el Html element to register the event habdler for
```
Comment (testflyjets) [inline src-ts/lib/functions/component-visible-functions/component-visible.functions.tsx:5]: Typo in "event habdler"

File: src-ts/lib/functions/component-visible-functions/component-visible.functions.tsx [hunk 2]
```diff
@@ -1,28 +1,53 @@
...
+    onMouseEnter: MouseEventHandler<HTMLDivElement>
+    onMouseLeave: MouseEventHandler<HTMLDivElement>
+}
 
-    function onClick(event: globalThis.MouseEvent): void {
-        setIsComponentVisible(!!ref.current?.contains(event.target))
-    }
+/**
+ * Registers an outside click event handler, and calls the callback on click outside
+ * @param el Html element to register the event habdler for
```
Comment (testflyjets) [inline src-ts/lib/functions/component-visible-functions/component-visible.functions.tsx:35]: Typo in "habdler" (copy paste typo?)

File: src-ts/lib/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -1,57 +1,73 @@
...
-    }: ComponentVisible = useHideClickOutside(false)
+    const triggerRef: MutableRefObject<any> = useRef(undefined)
+    const [tooltipOpen, setTooltipOpen]: [boolean, Dispatch<SetStateAction<boolean>>] = useState<boolean>(false)
 
-    function toggleOpen(): void {
-        const toggleTo: boolean = !tooltipOpen
-        setTooltipOpen(toggleTo)
-        setIsComponentVisible(toggleTo)
-    }
+    const toggleOpen: (t?: boolean) => void = useCallback((t?: boolean) => {
```
Comment (testflyjets) [inline src-ts/lib/tooltip/Tooltip.tsx:43]: Can we do better than one-letter variable names here? I'm used to seeing `<T>` for a generic type, but that's definitely an exception to the rule. What can we use that's better than `t` and `d` here?

Comment (brooketopcoder) [reply src-ts/lib/tooltip/Tooltip.tsx:43]: good call. maybe something like `toggledOpen`?

Comment (testflyjets) [reply src-ts/lib/tooltip/Tooltip.tsx:43]: Sure, that works -- and here I was thinking `t` was some sort of type placeholder. Even better that we fix it now that I know. Feel free to fix `d` while you're there!

Comment (vas3a) [reply src-ts/lib/tooltip/Tooltip.tsx]: Hey @testflyjets @brooketopcoder ! I updated the variable names, hope these are clear now. Thank you!

**Pull request merged**